cmd/query/app/querysvc/adjuster/otelattribute.go

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) 2024 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package adjuster
+
+import (
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+
+	"github.com/jaegertracing/jaeger/pkg/otelsemconv"
+)
+
+var otelLibraryKeys = map[string]struct{}{
+	string(otelsemconv.TelemetrySDKLanguageKey):   {},
+	string(otelsemconv.TelemetrySDKNameKey):       {},
+	string(otelsemconv.TelemetrySDKVersionKey):    {},
+	string(otelsemconv.TelemetryDistroNameKey):    {},
+	string(otelsemconv.TelemetryDistroVersionKey): {},
+}
+
+// OTELAttribute creates an adjuster that removes the OpenTelemetry library attributes
+// from spans and adds them to the attributes of a resource.
+func OTELAttribute() Adjuster {
+	return Func(func(traces ptrace.Traces) (ptrace.Traces, error) {
+		adjuster := otelAttributeAdjuster{}
```

Comment:
it feels rather convoluted, having a Func then a struct. Why not just return the struct from the public function and have it implement interface (and any helper methods) directly?

Reply:
Done - I'll make the same change to the other adjusters in a follow-up PR